feat(perf): speed up construction of bbjs Frs & cache zero hashes in ephemeral trees

dbanks12 opened the pull request:
Improves time for tx-hash computation from ~26-35ms down to ~20-23ms. This is because we were doing slower conversion of 1800+ Uint8Arrays to Frs. Allowing them to be actual Buffers lets us use `read/writeBigUint64` functions which are faster than the per-byte shifting we were doing.

Cache zero-hashes in static member of ephemeral tree class. Brings Ephemeral forest construction from 20ms down to 10ms.

Commit:
feat(perf): speed up construction of bbjs Frs

dbanks12 commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/11851?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#11851** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/11851?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/11851?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#11850** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/11850?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

fcarreiro commented on barretenberg/ts/src/types/fields.ts:
```diff
@@ -15,9 +15,10 @@ export class Fr {
   value: Uint8Array;
 
-  constructor(value: Uint8Array | bigint) {
+  constructor(value: Uint8Array | bigint | Buffer) {
```
Watch out, I think Buffers _are_ Uint8Arrays, and Buffer is only supported in Node (vs Uint8Array also in the browser)

dbanks12 replied:
Ugh.... This brings tx hash computation from 40ms down to 25ms

dbanks12 replied:
@Thunkar @ludamad is this why boxes test is failing?

ludamad replied:
No, i disabled vite tests in master

dbanks12 committed:
feat(perf): cleanup and minor speedups in AVM Ephemeral trees (#11852)

Cache zero-hashes in static member of ephemeral tree class. Brings
Ephemeral forest construction from 20ms down to 10ms.

Commit:
revert toBuffer  changes to try and fix boxes test

Commit:
Merge branch 'master' into db/ts-fr-hash-opts

Commit:
add back in toBuffer
